### PR TITLE
WICKET-5910 Do not intercept the protected methods with CGLib.

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: wicket-ioc/src/main/java/org/apache/wicket/proxy/LazyInitProxyFactory.java [hunk 1]
```diff
@@ -157,14 +164,26 @@ else if (type.isInterface())
 		}
 		else
 		{
-			CGLibInterceptor handler = new CGLibInterceptor(type, locator);
+			CGLibInterceptor handler = new CGLibInterceptor(type, locator) {
+				@Override
+				public Object intercept(Object object, Method method, Object[] args, MethodProxy proxy)
+						throws Throwable {
+					// TODO Auto-generated method stub
+					return super.intercept(object, method, args, proxy);
```
Comment (martin-g) [inline wicket-ioc/src/main/java/org/apache/wicket/proxy/LazyInitProxyFactory.java]: What is the purpose of this override? It seems to just call super.

File: wicket-ioc/src/main/java/org/apache/wicket/proxy/LazyInitProxyFactory.java [hunk 2]
```diff
@@ -349,6 +368,33 @@ public Object writeReplace() throws ObjectStreamException
...
+	{
+		private static final long serialVersionUID = 1L;
+	}
+
+	/**
+	 * CGLib callback filter: does not intercept protected methods.
+	 * 
+	 * Otherwise CGLib throws an IllegalArgumentException: protected method.
+	 */
+	private static class CGLibCallbackFilter implements CallbackFilter {
```
Comment (martin-g) [inline wicket-ioc/src/main/java/org/apache/wicket/proxy/LazyInitProxyFactory.java]: Please add a link to the description of the problem with protected methods in the Javadoc.

File: wicket-ioc/src/main/java/org/apache/wicket/proxy/LazyInitProxyFactory.java (hunk 1 above)
Comment (gsmet) [inline wicket-ioc/src/main/java/org/apache/wicket/proxy/LazyInitProxyFactory.java]: Urgh, sorry, it was a leftover of my work on it. Shouldn't have been there.

**Comment by gsmet** [comment]
I think I addressed your concerns.

**Pull request merged**